server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -229,6 +231,12 @@ public ClusterState execute(ClusterState currentState) {
                             metadata.getSettings()
                         ).ifPresent(validationErrors::add);
 
+                        if (FeatureFlags.isEnabled(FeatureFlags.READER_WRITER_SPLIT_EXPERIMENTAL_SETTING)) {
+                            validateAutoExpandReplicaConflictInRequest(normalizedSettings).ifPresent(validationErrors::add);
+                            validateAutoExpandReplicaConflictWithIndex(normalizedSettings, metadata.getSettings()).ifPresent(
```

Comment:
lets combine these into a single call, `validateAutoExpandReplicaConflictInRequest` can be called from `validateAutoExpandReplicaConflictWithIndex`